[KB-272] Initial results - table generation and analysis (#198)
chhoumann/thesis-chemcam: report_thesis/src/sections/results/initial_results.tex

\subsection{Initial Results}
As described in Section~\ref{sec:proposed_approach}, we conducted a series of initial experiments to evaluate the performance of various machine learning models on the prediction of major oxide compositions from our \gls{libs} dataset.
These experiments aimed to provide a preliminary assessment of the models' performance.
All models were trained on the same preprocessed data using the Norm 3 preprocessing method described in Section~\ref{sec:norm3}.

> This ensured that the models' performance could be evaluated under consistent and comparable conditions.

Reviewer: You didn't know this, but I actually explained this detail in the experiment design. What do you think? Is it okay to summarize the point again?

Reviewer 2: I really like Patrick's argument, but think it should be put in experiment design only. The results section is only for showing results & discussing, not presenting our approach.

Can we move it to the experiment design? And what are your thoughts @Pattrigue @Ivikhostrup?

Reviewer 2: Perhaps a _very_ brief summary is fine to have, but not much more. The first few lines provide such a (nice) summary, and it feels like to have a 'refresher'.

But I think we need to take a pass over the sections we ref back to at some point. This one refs back to proposed approach, which is OK, but we also have a section regarding experiment design for this experiment. That should probably also get a ref, no?

Reviewer 2: Would still like to hear thoughts on this + can someone follow up to ensure this gets done? @Pattrigue @Ivikhostrup

Reviewer: This is the argument I used in the experiment design:

To conduct the initial experiment and evaluate the twelve selected models shown in Table~\ref{tab:preprocessing-models}, we ensured that the models were evaluated on a consistent basis.
We achieve this by setting the following conditions for the experiment.
The only preprocessor used for all models was \texttt{Norm 3} normalization.
This normalization technique was developed specifically by \citet{cleggRecalibrationMarsScience2017} for accomodating the characteristics of the \gls{libs} data, whereby the data represents the readings of three different spectrometers, as outlined in Section~\ref{sec:norm3}.
Therefore, \texttt{Norm 3} represents the most \gls{libs}-specific normalization technique available in from the pool of selected preprocessors and is the most appropriate choice for this experiment.

To me this sounds similar to what patrick has written here. We could take Patricks wording and replace what I have written if you want.

Reviewer: In terms of referring back to the initial experiments design, I'd have to look again, but it would make sense to do so.

Reviewer 2: Need to ensure coherence between this section and the experiment design. E.g., we don't ref back, but make the exact same point about norm 3. Defeats the purpose if we explain it again.

Just do what makes most sense here, would prefer not having to mediate that process

Table~\ref{tab:init_results} presents the results of these experiments, including the \gls{rmsep}, \gls{rmsecv}, standard deviation, standard deviation of the cross-validation, and the mean of these metrics for each model and oxide.
The primary metric of interest is the \gls{rmsecv}, as it provides a measure of the model's predictive accuracy during cross-validation, reflecting its generalization performance on unseen data.
Additionally, the standard deviation of the cross-validation offers insight into the variability of the model's performance across different folds, and the \gls{rmsep} indicates the model's predictive accuracy on a separate test set.

Ridge regression and \gls{svr} generally exhibit strong performance across different oxides, as evidenced by their relatively low \gls{rmsecv} values.
In contrast, \gls{enet}, \gls{ann}, and \gls{cnn} tend to perform the worst, with higher \gls{rmsecv} values.

> Models such as \gls{rf} and \gls{gbr} exhibit moderate performance, often surpassing simpler linear models but not as consistently as \gls{svr}.

Reviewer 2: RF is indeed moderate, but GBR is top 3. 

And this is phrased a bit weird because we consider SVR a linear model.
> often surpassing simpler linear models but not as consistently as \gls{svr}.

For $\ce{SiO2}$, \gls{svr} demonstrates the lowest \gls{rmsecv} of 4.908.
Ridge regression and \gls{rf} also perform well, with \gls{rmsecv} values of 5.005 and 5.304, respectively.

> The standard deviations for these models are relatively low, indicating stable performance.

Reviewer 2: Again, I'm a bit confused as to why these exact models were selected? Picking top 2, 4, and 7?

![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/6501c8dc-3978-4db1-9596-bf4676e3e6f0)

> In the case of $\ce{TiO2}$, \gls{etr} exhibits the lowest \gls{rmsecv} of 0.427, signifying exceptional performance, with \gls{svr} and \gls{ngboost} also showing strong predictive capabilities, with \gls{rmsecv} values of 0.463 and 0.433, respectively.

Reviewer: "Signifying exceptional performance" might be a little dangerous to say when the reader does not know what we are comparing against.

If we do decide to compare it against, say the MOC model, we have to be very careful that it doesnt sound like MOC is the baseline. Perhaps we can compare it to supercam?

They got 0.3 RMSEP with RF, so maybe we can discuss how its actually pretty impressive to get 0.427 considering the CV folds have all the extreme values and we have done zero tuning (Obviously, our choice of parameters might just be a lucky pick as well, but we can verify this later when we actually get to the hyperparameter tuning)

Author: ```suggestion
In the case of $\ce{TiO2}$, \gls{etr} exhibits the lowest \gls{rmsecv} of 0.427, indicating very good performance, with \gls{svr} and \gls{ngboost} also showing strong predictive capabilities, with \gls{rmsecv} values of 0.463 and 0.433, respectively.
```

> In the case of $\ce{TiO2}$, \gls{etr} exhibits the lowest \gls{rmsecv} of 0.427, signifying exceptional performance, with \gls{svr} and \gls{ngboost} also showing strong predictive capabilities, with \gls{rmsecv} values of 0.463 and 0.433, respectively.

Reviewer 2: You mean Random Forest here, right? You use the number for RF, but write ETR.

No idea why you write about SVR, though. It's top 6. NGBoost is top 2, so it's good that you highlight it.

> he low standard deviation of the cross-validation for these models indicates consistent performance across folds.

Reviewer: ```suggestion
The low standard deviation of the cross-validation for these models indicates consistent performance across folds.
```

Reviewer: I didnt comment on this earlier, but I think I need to add to the experiment design what signifies low and high rmscv and std. dev.
This would make your argument stronger here, by saying something like:" Since its below x, then we deem that this is a consistent performance"

what do you think? Is that redundant?

> For $\ce{Al2O3}$, \gls{ngboost} with an \gls{rmsecv} of 2.291 and ridge regression with an \gls{rmsecv} of 3.115 achieve relatively low \gls{rmsecv}, highlighting their effectiveness in predicting this oxide, with \gls{svr} also performing well with an \gls{rmsecv} of 2.700.

Reviewer 2: OK, I'll stop reading at this point, because you consistently highlight the wrong models as having the top performance. I've doubled checked in the data & in the table on those I've already mentioned.

It doesn't make sense to me why we would highlight inferior performance as being good. Please let me know your thoughts, as I'm super confused by this.

> he standard deviation analysis reveals that \gls{ngboost} has the least variability, making it a more reliable model for this oxide.

Reviewer: ```suggestion
The standard deviation analysis reveals that \gls{ngboost} has the least variability, making it a more reliable model for this oxide.
```

Reviewer: Regarding the comment above: 
This might actually be a better way to approach it, since setting some arbitrary threshold is a little iffy. I like this where you say it has the least variability, therefore..."

\gls{svr} emerges as the best-performing model for $\ce{FeO_T}$ with the lowest \gls{rmsecv} of 2.847, while ridge regression with an \gls{rmsecv} of 3.385 and \gls{ngboost} with an \gls{rmsecv} of 3.561 also demonstrate competent performance.
The standard deviation values indicate that while \gls{svr} has lower variability, the other models also maintain reasonable consistency.

For $\ce{MgO}$, \gls{svr} exhibits the best performance with the lowest \gls{rmsecv} of 1.426, and ridge regression with an \gls{rmsecv} of 1.509 and \gls{ngboost} with an \gls{rmsecv} of 1.578 are also effective.

> The standard deviation metrics suggest that these models are stable, with \gls{svr} showing the least variation.

Reviewer: ```suggestion
The standard deviation metrics suggest that these models are more stable, with \gls{svr} showing the least variation.
```
If we want to use the same approach as with ngboost

The standard deviation analysis supports these findings, indicating reliable performance for these models.

To summarize, \gls{svr} stands out as the best overall model based on its average performance, with a mean \gls{rmsecv} of 1.958. This is closely followed by \gls{ngboost} with a mean \gls{rmsecv} of 2.017 and ridge regression with a mean \gls{rmsecv} of 2.111.
When examining the \gls{rmsep} values, \gls{svr} also demonstrates strong predictive accuracy on independent test sets, reinforcing its effectiveness.
Additionally, the standard deviation of the cross-validation for \gls{svr} is generally low, indicating stable and consistent performance across different subsets of data.

\gls{ngboost} and ridge regression also show robust performance, with relatively low standard deviations.
Specifically, \gls{ngboost} exhibits consistent performance with low variability, making it a dependable model for various oxides.
Ridge regression, while slightly higher in \gls{rmsecv}, still maintains good performance with low standard deviation values.

> Overall, \gls{svr}, ridge regression, and \gls{ngboost} are consistently among the top-performing models for multiple oxides, as evidenced by their low \gls{rmsecv}, \gls{rmsep}, and standard deviation values.

Reviewer: I like this summary. However, I think what is missing is a conclusion on what models to proceed with per oxide.

For example since SVR is such a high performer its naturally on to the next round (could be used for all oxides), but do this for all the other models for each oxide.
Just as an aside - Whatever choice you make here is perfectly valid, so long as its based in their performance. Just make sure we have like 5-7 ish models that make it on the next round.

Reviewer: and that they are diverse. So a good blend from each category would be preferable. (This is the argument we made in models selection - Stacking needs diverse models for each oxide)

Reviewer 2: I can't agree with this summary. It doesn't match what I can see in the data nor the table.

(but I agree with SVR being good, it's top 2 on avg. performance measured by RMSECV!)

And why would 5-7 models 'make it to the next round' when we've obviously chosen 10? We ran the experiments for 10. See my comment on disqualifying only the worst 3, as their performance shows a steep increase in RMSECV compared to the other models.

\subsection{Initial Results}
As described in Section~\ref{sec:proposed_approach}, we conducted a series of initial experiments to evaluate the performance of various machine learning models on the prediction of major oxide compositions from our \gls{libs} dataset.

> These experiments aimed to provide a preliminary assessment of the models' performance.

Author: ```suggestion
These experiments aimed to provide a preliminary assessment of the models' performance, allowing us to identify the most promising models for further evaluation and inclusion in our stacking ensemble.
```

> These experiments aimed to provide a preliminary assessment of the models' performance.

Reviewer 2: Can't commit, says outdated.

All models were trained on the same preprocessed data using the Norm 3 preprocessing method described in Section~\ref{sec:norm3}.
This ensured that the models' performance could be evaluated under consistent and comparable conditions.

> Table~\ref{tab:init_results} presents the results of these experiments, including the \gls{rmsep}, \gls{rmsecv}, standard deviation, and standard deviation of cross-validation prediction errors for each model across all oxides.

Reviewer 2: @Pattrigue can you fix?

> Table~\ref{tab:init_results} presents the results of these experiments, including the \gls{rmsep}, \gls{rmsecv}, standard deviation, and standard deviation of cross-validation prediction errors for each model across all oxides.

Author: Should be fixed now.

The means of each metric are also provided to give an overall indication of the models' performance.
Furthermore, we present an overview of these mean values in Figure~\ref{fig:init_results_rmses} to facilitate a visual comparison of the models' general performance.

> The results indicate that the gradient boosting models, \gls{xgboost}, \gls{gbr}, and \gls{ngboost}, consistently perform well across all oxides, with \gls{xgboost} generally outperforming the other two gradient boosting models.

Reviewer 2: @Pattrigue mention this or no?

> The results indicate that the gradient boosting models, \gls{xgboost}, \gls{gbr}, and \gls{ngboost}, consistently perform well across all oxides, with \gls{xgboost} generally outperforming the other two gradient boosting models.

Author: I'm not sure this would bring anything significant to the report but maybe we could say something like this?

```suggestion
The results indicate that the gradient boosting models, \gls{xgboost}, \gls{gbr}, and \gls{ngboost}, consistently perform well across all oxides, with \gls{xgboost} generally outperforming the other two gradient boosting models.
Interestingly, \gls{gbr} has the lowest \gls{rmsep}, while \gls{xgboost} achieves the lowest \gls{rmsecv}, suggesting that the regularization in \gls{xgboost} may improve the model's generalizability.
```

We observe that \gls{enet} performs between ridge and \gls{lasso} in terms of both error and standard deviation, which aligns with expectations since \gls{enet} combines the regularization techniques of both models.

The \gls{cnn} and \gls{ann} models perform the worst across all oxides, exhibiting the highest mean \gls{rmsep} and \gls{rmsecv} values, as well as the highest standard deviation values.
This poor performance is further highlighted in Table~\ref{tab:relative_performance}, which shows the relative performance of each model compared to the best-performing model, \gls{xgboost}.
The table also includes the difference in performance relative to the next best model, with \gls{xgboost} serving as the baseline for comparison, assigned a relative performance of 100\%.
From this table, it is evident that the \gls{cnn} and \gls{ann} models experience significant drops in performance compared to the top-performing models.
While deep learning models such as these have the theoretical potential to perform well with \gls{libs} data, given their ability to learn complex patterns and relationships, the relatively small size of our dataset may limit their efficacy.

> In addition, this would require more effective tuning of hyperparameters, which is challenging due to the vast number of possible permutations.

Author: Rephrased this

> Hyperparameter tuning akin to the method detailed in Section~\ref{sec:optimization_framework} could be pursued, but the required computational time would be prohibitive.

Author: Good point, included this point in the argument now.

For these reasons, we decided to exclude the \gls{cnn} and \gls{ann} models from further experimentation.

Tables~\ref{tab:best_results} and \ref{tab:best_model_occurrences} list the best-performing model for each oxide and the frequency with which each model achieves top performance according to various metrics, respectively.
These tables are intended to provide an overview of model performance rather than to determine an overall 'winner by majority'.
Their purpose is to illustrate the general trends and behavior of different models across various metrics and oxides.

> Although \gls{xgboost} and \gls{svr} appear the most frequently in Table~\ref{tab:best_model_occurrences}, this does not imply that they are the best models for every oxide.
> The tables also demonstrates the importance of judging model performance per oxide.
> For example, if one were to only consider the mean of the performance metrics, \gls{pls} would be considered among the worst performing models, as shown in Figure~\ref{fig:init_results_rmses}.
> However, inspecting Table~\ref{tab:best_results} reveals that \gls{pls} exhibits the lowest \gls{rmsecv} and standard deviation of prediction errors for both \ce{MgO} and \ce{Na2O}, indicating that it is the most accurate and robust model for these oxides.

Author: ```suggestion
Although \gls{xgboost} and \gls{svr} appear the most frequently in Table~\ref{tab:best_model_occurrences}, this does not imply that they are the best models for every oxide.
The tables also demonstrates the importance of judging model performance per oxide.
For example, if one were to only consider the mean of the performance metrics, \gls{pls} would be considered among the worst performing models, as shown in Figure~\ref{fig:init_results_rmses}.
However, inspecting Table~\ref{tab:best_results} reveals that \gls{pls} exhibits the lowest \gls{rmsecv} and standard deviation of prediction errors for both \ce{MgO} and \ce{Na2O}, indicating that it is the most accurate and robust model for these oxides and supporting the necessity of evaluating model performance on a per-oxide basis, as discussed in Section~\ref{sec:proposed_approach}.
```

\begin{figure*}[h]
    \centering
    \includegraphics[width=\textwidth]{images/init_results_means.png}
    \caption{Mean \gls{rmsep}, \gls{rmsecv}, standard deviation of prediction errors, and standard deviation of cross-validation prediction errors for each model across all oxides.}
    \label{fig:init_results_rmses}
\end{figure*}

\begin{table}[ht]
\centering
\begin{tabularx}{\linewidth}{lrr}
\toprule
Model & Relative Performance (\%) & Diff. vs Next (\%) \\
\midrule
\gls{xgboost} & 100.00 & 0.85 \\
\gls{svr} & 100.85 & 2.22 \\
\gls{gbr} & 103.07 & 0.87 \\
\gls{ngboost} & 103.94 & 0.51 \\
\gls{rf} & 104.45 & 0.39 \\
\gls{etr} & 104.84 & 3.89 \\
Ridge & 108.74 & 2.92 \\
\gls{pls} & 111.66 & 2.73 \\
\gls{lasso} & 114.38 & 13.44 \\
\gls{ann} & 127.82 & 15.36 \\
\gls{cnn} & 143.18 & 39.18 \\
\gls{enet} & 182.36 & - \\
\bottomrule
\end{tabularx}
\caption{Relative performance of each model compared to the best performing model, and the difference in performance compared to the next best model.}
\label{tab:relative_performance}

> \end{table}

Author: Forgot to update the table, good catch, should now be fixed.

\gls{svr} also performed well, with similar accuracy and robustness to the gradient boosting models.
In contrast, deep learning models such as \gls{cnn} and \gls{ann} underperformed, likely due to the small dataset size and insufficient tuning of their architectures and hyperparameters.
Inspecting the model performances per oxide revealed that the best model varied depending on the oxide, and several models performed well for each oxide.
This emphasizes the need for further evaluation of model performances on a per-oxide basis to identify suitable configurations for our stacking ensemble approach, which aims to leverage the strengths of multiple models.

\begin{figure*}[h]
    \centering
    \includegraphics[width=\textwidth]{images/init_results_means.png}
    \caption{Mean \gls{rmsep}, \gls{rmsecv}, standard deviation of prediction errors, and standard deviation of cross-validation prediction errors for each model across all oxides.}
    \label{fig:init_results_rmses}
\end{figure*}

\begin{table}[ht]
\begin{tabular}{lrr}
\toprule

> Model & Relative Performance (\%) & Diff. vs Prev. (\%) \\

Reviewer 2: ```suggestion
Model & Relative Performance (\%) & Diff. vs Prev. \\
```

This ensured that the models' performance could be evaluated under consistent and comparable conditions.

Table~\ref{tab:init_results} presents the results of these experiments, including the \gls{rmsep}, \gls{rmsecv}, standard deviation, and standard deviation of cross-validation prediction errors for each model across all oxides.
The means of each metric are also provided to give an overall indication of the models' performance.
Furthermore, we present an overview of these mean values in Figure~\ref{fig:init_results_rmses} to facilitate a visual comparison of the models' general performance.

The results indicate that the gradient boosting models, \gls{xgboost}, \gls{gbr}, and \gls{ngboost}, consistently perform well across all oxides, with \gls{xgboost} generally outperforming the other two gradient boosting models.
These models exhibit both low mean \gls{rmsep} and \gls{rmsecv} values, indicating high accuracy, as well as low standard deviation values, underscoring their robustness.
\gls{svr} is also among the top-performing models, with mean \gls{rmsep} and \gls{rmsecv} values close to those of \gls{xgboost} and low standard deviation values.

> While usually outperformed by gradient boosting models and \gls{svr}, the ensemble models, \gls{rf} and \gls{etr}, also exhibit good performance.

Reviewer 2: ahh we need to be careful, i just remembered an issue in the model selection section i forgot to point out a few days ago

gradient boosting models are also ensemble models

(issue in table is: we call the linear models 'regression' models - they all are! I'll fix this)

Reviewer 2: so can you correct your use of `ensemble` here?

Reviewer 2: here's what I mean

![image](https://github.com/chhoumann/thesis-chemcam/assets/29108628/33ce2add-4353-444e-be3d-7dec3e372df9)

Author: Unsure what term you want me to use then, could you make a suggestion to ensure it aligns with the table fix?

Reviewer 2: Yeah, the table thing was just a tangent because I remembered it and wanted to let you know.

The issue here is that it sounds like we don't recognize the GB models as ensembles, which they are.
Just do something like
```suggestion
While usually outperformed by gradient boosting models and \gls{svr}, other ensemble models, such as \gls{rf} and \gls{etr}, also exhibit good performance.
```

Reviewer 2: But probably not `such as`

Author: ```suggestion
While usually outperformed by gradient boosting models and \gls{svr}, other ensemble models, including \gls{rf} and \gls{etr}, also exhibit good performance
```

The \gls{pls}, ridge, \gls{lasso}, and \gls{enet} models typically seem to perform worse than the other models, with higher mean \gls{rmsep} and \gls{rmsecv} values and higher standard deviation values.
We observe that \gls{enet} performs between ridge and \gls{lasso} in terms of both error and standard deviation, which aligns with expectations since \gls{enet} combines the regularization techniques of both models.

The \gls{cnn} and \gls{ann} models perform the worst across all oxides, exhibiting the highest mean \gls{rmsep} and \gls{rmsecv} values, as well as the highest standard deviation values.
This poor performance is further highlighted in Table~\ref{tab:relative_performance}, which shows the relative performance of each model compared to the best-performing model, \gls{xgboost}.
The table also includes the difference in performance relative to the next best model, with \gls{xgboost} serving as the baseline for comparison, assigned a relative performance of 100\%.

> From this table, it is evident that the \gls{cnn} and \gls{ann} models experience significant drops in performance compared to the top-performing models.

Reviewer 2: Need to be careful with significant unless it's statistically significant we mean

Reviewer 2: Easily misunderstood. So just use another word?

Author: ```suggestion
From this table, it is evident that the \gls{cnn} and \gls{ann} models experience notable drops in performance compared to the top-performing models.
```

While deep learning models such as these have the theoretical potential to perform well with \gls{libs} data, given their ability to learn complex patterns and relationships, the relatively small size of our dataset may limit their efficacy.

> Furthermore, achieving optimal performance with these models necessitates extensive hyperparameter tuning, which involves exploring a vast space of potential configurations and design choices.

Reviewer 2: Just to clarify: hyperparameters are things like learning rate. Ensuring a good architecture is also _very_ time-consuming. Hyperparameters might actually be comparatively easy because there are good rules of thumb to follow.

```suggestion
Furthermore, achieving optimal performance with these models necessitates extensive tuning of both their architectures and hyperparameters, which involves exploring a vast space of potential configurations and design choices.
```

Reviewer 2: (* For DL)

lthough methods for systematic hyperparameter optimization, as detailed in Section~\ref{sec:optimization_framework}, could be employed, the associated computational cost would be prohibitively high.
Additionally, there are numerous architectural design decisions and advanced techniques that could potentially enhance model performance, but their inclusion would significantly expand the scope of this study beyond feasible limits.
For these reasons, we decided to exclude the \gls{cnn} and \gls{ann} models from further experimentation.

Tables~\ref{tab:best_results} and \ref{tab:best_model_occurrences} list the best-performing model for each oxide and the frequency with which each model achieves top performance according to various metrics, respectively.
These tables are intended to provide an overview of model performance rather than to determine an overall 'winner by majority'.
Their purpose is to illustrate the general trends and behavior of different models across various metrics and oxides.
Although \gls{xgboost} and \gls{svr} appear the most frequently in Table~\ref{tab:best_model_occurrences}, this does not imply that they are the best models for every oxide.

> The tables also demonstrates the importance of judging model performance per oxide.

Reviewer 2: This sentence actually feels like it kind of 'stops' the argument to restate the point you're trying to make, only to proceed and confirm it after. 

Does not imply they're the best for all oxides → For example → However, inspecting, we see that these metrics for X model is low on YZ oxides → So need to judge on a per-oxide basis

That's perfect. So just remove the one sentence that breaks that flow. Feels like an aside, but is actually directly related. So it's superfluous.

```suggestion
```

For example, if one were to only consider the mean of the performance metrics, \gls{pls} would be considered among the worst performing models, as shown in Figure~\ref{fig:init_results_rmses}.
However, inspecting Table~\ref{tab:best_results} reveals that \gls{pls} exhibits the lowest \gls{rmsecv} and standard deviation of prediction errors for both \ce{MgO} and \ce{Na2O}.

This indicates that \gls{pls} is the most accurate and robust model for these oxides, underscoring the importance of evaluating model performance on a per-oxide basis, as discussed in Section~\ref{sec:proposed_approach}.
Moreover, for some oxides, multiple models perform similarly well, such as \gls{xgboost}, \gls{gbr}, and ridge for \ce{CaO}, which further emphasizes the need to leverage the strengths of multiple models.

To summarize, the initial results indicate that gradient boosting models, particularly \gls{xgboost}, demonstrated the most consistent and accurate performance across all oxides.
\gls{svr} also performed well, with similar accuracy and robustness to the gradient boosting models.

> In contrast, deep learning models such as \gls{cnn} and \gls{ann} underperformed, likely due to the small dataset size and insufficient hyperparameter tuning.

Reviewer 2: ```suggestion
In contrast, deep learning models such as \gls{cnn} and \gls{ann} underperformed, likely due to the small dataset size and insufficient tuning of their architectures and hyperparameters.
```

The table also includes the difference in performance relative to the next best model, with \gls{xgboost} serving as the baseline for comparison, assigned a relative performance of 100\%.
From this table, it is evident that the \gls{cnn} and \gls{ann} models experience significant drops in performance compared to the top-performing models.
While deep learning models such as these have the theoretical potential to perform well with \gls{libs} data, given their ability to learn complex patterns and relationships, the relatively small size of our dataset may limit their efficacy.
Furthermore, achieving optimal performance with these models necessitates extensive hyperparameter tuning, which involves exploring a vast space of potential configurations and design choices.
lthough methods for systematic hyperparameter optimization, as detailed in Section~\ref{sec:optimization_framework}, could be employed, the associated computational cost would be prohibitively high.

> Additionally, there are numerous architectural design decisions and advanced techniques that could potentially enhance model performance, but their inclusion would significantly expand the scope of this study beyond feasible limits.

Reviewer 2: filler word imo
```suggestion
Additionally, there are numerous architectural design decisions and advanced techniques that could potentially enhance model performance, but their inclusion would expand the scope of this study beyond feasible limits.
```

Although \gls{xgboost} and \gls{svr} appear the most frequently in Table~\ref{tab:best_model_occurrences}, this does not imply that they are the best models for every oxide.
The tables also demonstrates the importance of judging model performance per oxide.
For example, if one were to only consider the mean of the performance metrics, \gls{pls} would be considered among the worst performing models, as shown in Figure~\ref{fig:init_results_rmses}.
However, inspecting Table~\ref{tab:best_results} reveals that \gls{pls} exhibits the lowest \gls{rmsecv} and standard deviation of prediction errors for both \ce{MgO} and \ce{Na2O}.

> 

Reviewer 2: Probably don't want  the linebreak when you're referencing the oxides in the sentence below?
```suggestion
```

This indicates that \gls{pls} is the most accurate and robust model for these oxides, underscoring the importance of evaluating model performance on a per-oxide basis, as discussed in Section~\ref{sec:proposed_approach}.

> Moreover, for some oxides, multiple models perform similarly well, such as \gls{xgboost}, \gls{gbr}, and ridge for \ce{CaO}, which further emphasizes the need to leverage the strengths of multiple models.

Reviewer 2: Is that really what that implies? I'm not so sure, but let me know your thoughts.

Also: we'd be assuming that the models don't make the same kinds of errors, which isn't clear yet.

Author: Here is a more cautious way to phrase it - thoughts? Thought we could use it to lead into the stacking ensemble section.
```suggestion
Moreover, for some oxides, multiple models perform similarly well, such as \gls{xgboost}, \gls{gbr}, and ridge for \ce{CaO}.
This observation suggests the potential benefit of leveraging the strengths of multiple models, provided they do not make similar types of errors, which warrants further investigation.
```

Reviewer 2: Yeah, that's good, but could you make a task to follow up?

\end{figure*}

\begin{table}[ht]
\begin{tabularx}{lrr}
\toprule
Model & Relative Performance (\%) & Diff. vs Next (\%) \\
\midrule
XGB & 100.00 & - \\
SVR & 100.85 & 0.85 \\
GBR & 103.07 & 2.22 \\
NGB & 103.94 & 0.87 \\
RandomForest & 104.45 & 0.51 \\
ExtraTrees & 104.84 & 0.39 \\
Ridge & 105.04 & 0.20 \\
PLS & 111.66 & 6.61 \\
ElasticNet & 114.12 & 2.46 \\
LASSO & 114.30 & 0.19 \\
ANN & 127.82 & 13.52 \\
CNN & 143.18 & 15.36 \\
\bottomrule
\end{tabularx}

> \caption{Relative performance of each model compared to the best performing model, and the difference in performance compared to the next best model.}

Reviewer 2: A few points on this one:
- Diff vs Prev. (%) is misleading because it's measured in percentage points / actual values, not percentages
- Need to clarify performance is measured by RMSECV, and that a higher percentage is worse
- Might need text clarifying what's shown ('normalized' RMSECV * 100 for percentage, with the best performing as the 'baseline'). But it's kinda covered in the 'assigned a relative performance of 100%'

Author: ```suggestion
\caption{Relative performance of each model compared to the best performing model, measured by normalized RMSECV (multiplied by 100 for percentage). A higher percentage indicates worse performance. The 'Diff. vs Prev.' column shows the difference in performance compared to the next best model, measured in percentage points.}
```

Author: > Diff vs Prev. (%) is misleading because it's measured in percentage points / actual values, not percentages

I can't really think of better column names that would encapsulate this without being far too long. It barely fits as is. Any ideas?

Reviewer 2: Drop `(%)`  and just have `Diff vs. Prev.`?
